src/internal/client.ts

```diff
@@ -1031,4 +1032,92 @@ export class TypedClient {
     const xmlResult = await readAsString(httpRes)
     return xmlParsers.parseReplicationConfig(xmlResult)
   }
+
+  getObjectLegalHold(
+    bucketName: string,
+    objectName: string,
+    getOpts?: GetObjectLegalHoldOptions,
+    callback?: ResultCallback<LegalHoldStatus>,
+  ): Promise<LegalHoldStatus>
```

Comment:
```suggestion
  getObjectLegalHold(
    bucketName: string,
    objectName: string,
    getOpts?: GetObjectLegalHoldOptions,
  ): void
  async getObjectLegalHold(
    bucketName: string,
    objectName: string,
    getOpts?: GetObjectLegalHoldOptions,
    callback?: ResultCallback<>,
  ): Promise<LegalHoldStatus>
```

should seprated to 2 methods, I think

Reply:
We decided to deprecate callback style and based on the earlier suggestion, i removed this. @kaankabalak @trim21

Reply:
in this case I think we should just remove `callback: ResultCallback<LegalHoldStatus>,`

Reply:
Thank you.